src/bullet.py

```diff
@@ -1,72 +1,112 @@
 from __future__ import annotations
 
-import math
-from typing import TYPE_CHECKING, List, cast
+from typing import TYPE_CHECKING, List, Tuple, cast
 
 import arcade
-
-from arena.wall import Wall
-from constants import SCREEN_HEIGHT, SCREEN_WIDTH
-from iron_math import add_vec, move_sprite_relative_to_parent, rotate_vec
+from iron_math import move_sprite_polar, set_sprite_location, sprite_in_bounds
 from linked_sprite import LinkedSprite
 from sprite_lists import SpriteLists
 
 # This allows a circular import only for the purposes of type hints
 # Weapon will never create and instance of Player
 if TYPE_CHECKING:
-    from weapon import Weapon
     from player import Player
 
 
 class Projectile:
-    def __init__(self, sprite: LinkedSprite[Projectile], sprite_lists: SpriteLists):
+    """
+    A projectile can be anything that is created by a weapon or another projectile included beams, rockets, etc.
+    It stores the characteristics of the projectile and is responible the associated sprite
+    """
+
+    sprite: LinkedSprite[Projectile]
+    sprite_lists: SpriteLists
+    damage: float
+    speed: float
+    radians: float
+    sprite_rotation_offset: float
+    exists: bool
```

Comment:
```suggestion
    exists: bool
    """
    Projectile is visible and can be collided with.  Used for long-lived projectile objects that are repeatedly added to/removed from the world over time, such as laser beams.
    """
```

Maybe slap a docstring onto this?  Since it's a bit obscure.

Reply:
Fixed